code/game/gamemodes/miniantags/guardian/guardian.dm

```diff
@@ -208,18 +211,23 @@
 	set category = "Guardian"
 	set desc = "Communicate telepathically with your guardian."
 	var/input = stripped_input(src, "Please enter a message to tell your guardian.", "Message", "")
-	if(!input) return
+	if(!input)
+		return
 
-	for(var/mob/M in GLOB.mob_list)
-		if(istype(M, /mob/living/simple_animal/hostile/guardian))
-			var/mob/living/simple_animal/hostile/guardian/G = M
-			if(G.summoner == src)
-				to_chat(G, "<span class='changeling'><i>[src]:</i> [input]</span>")
-				log_say("(GUARDIAN to [key_name(G)]) [input]", src)
+	// Find the guardian in our host's contents.
+	var/mob/living/simple_animal/hostile/guardian/G = locate() in contents
+	if(!G)
+		return
 
-		else if(M in GLOB.dead_mob_list && M.client && M.stat == DEAD && !isnewplayer(M))
-			to_chat(M, "<span class='changeling'><i>Guardian Communication from <b>[src]</b> ([ghost_follow_link(src, ghost=M)]): [input]</i>")
+	// Show the message to our guardian and to host.
+	to_chat(G, "<span class='changeling'><i>[src]:</i> [input]</span>")
 	to_chat(src, "<span class='changeling'><i>[src]:</i> [input]</span>")
+	log_say("(GUARDIAN to [key_name(G)]) [input]", src)
+
+	// Show the message to any ghosts/dead players.
+	for(var/mob/M in GLOB.dead_mob_list)
+		if(M && M.client && M.stat == DEAD && !isnewplayer(M))
```

Comment:
Same as the previous one. Is the dead check required?

Reply:
Just left it in because it was already there but.. its probably not needed if they're in the dead mob list, unless we have some fucky stuff about people not getting removed from it properly.

Reply:
There could have been an old bug where alive things were in the dead list. Not sure about this one myself. Maybe a maint might know a bit more
I don't mind you leaving it here just in case.

Reply:
I'll just leave it here and see what they think about it for now.

Reply:
Yeah, this is all kinds of messy--until we're 100% certain about the dead and alive mobs lists....I'm going to say this is probably ok for now.